src/utils.ts

```diff
@@ -45,22 +55,29 @@ const constructInteraction = (intercept: Interception | AnyObject, testTitle: st
     request: {
       method: intercept.request.method,
       path: path,
-      headers: intercept.request.headers,
+      headers: filterHeaders(intercept.request.headers, headerAllowList),
       body: intercept.request.body,
       query: query
     },
     response: {
       status: intercept.response?.statusCode,
-      headers: intercept.response?.headers,
+      headers: filterHeaders(intercept.response?.headers, headerAllowList),
       body: intercept.response?.body
     }
   }
 }
+
+export const getHeaderAllowlist = (headerAllowList: string[]) => {
+  const headerAllowlist = headerAllowList || Cypress.env('pactHeaderAllowlist') || []
```

Comment:
should this just be inlined rather than assigned and immediately returned? (I recall eslint used to inline that, but will leave it up to your judgment)